org.opentosca.bus.management.service.impl/src/org/opentosca/bus/management/service/impl/collaboration/DeploymentDistributionDecisionMaker.java

```diff
@@ -0,0 +1,420 @@
+
+        // create collaboration message
+        final BodyType content = new BodyType(new InstanceDataMatchingRequest(infrastructureNodeType, properties));
+        final CollaborationMessage collaborationMessage = new CollaborationMessage(new KeyValueMap(), content);
+
+        // create an unique correlation ID for the request
+        final String correlationID = UUID.randomUUID().toString();
+
+        DeploymentDistributionDecisionMaker.LOG.debug("Publishing instance matching request to MQTT broker at {} with topic {} and correlation ID {}",
+                                                      Constants.LOCAL_MQTT_BROKER, Constants.REQUEST_TOPIC,
+                                                      correlationID);
+
+        // create exchange containing the collaboration message and the needed headers
+        final Exchange request =
+            new ExchangeBuilder(Activator.camelContext).withBody(collaborationMessage)
+                                                       .withHeader(MBHeader.MQTTBROKERHOSTNAME_STRING.toString(),
+                                                                   Constants.LOCAL_MQTT_BROKER)
+                                                       .withHeader(MBHeader.MQTTTOPIC_STRING.toString(),
+                                                                   Constants.REQUEST_TOPIC)
+                                                       .withHeader(MBHeader.REPLYTOTOPIC_STRING.toString(),
+                                                                   Constants.RESPONSE_TOPIC)
+                                                       .withHeader(MBHeader.CORRELATIONID_STRING.toString(),
+                                                                   correlationID)
+                                                       .withHeader(MBHeader.REMOTEOPERATION_STRING.toString(),
+                                                                   RemoteOperations.invokeInstanceDataMatching)
+                                                       .build();
+
+        // publish the exchange over the camel route
+        final Thread thread = new Thread(() -> {
+
+            // By using an extra thread and waiting some time before sending the request, the
+            // consumer can be started in time to avoid loosing replies.
+            try {
+                Thread.sleep(300);
+            }
+            catch (final InterruptedException e) {
+            }
+
+            Activator.producer.send("direct:SendMQTT", request);
+        });
+        thread.start();
+
+        final String callbackEndpoint = "direct:Callback-" + correlationID;
+        DeploymentDistributionDecisionMaker.LOG.debug("Waiting for response at endpoint: {}", callbackEndpoint);
+
+        // wait for a response and consume it or timeout after 10s
+        final ConsumerTemplate consumer = Activator.camelContext.createConsumerTemplate();
+        final Exchange response = consumer.receive(callbackEndpoint, 10000);
```

Comment:
From sending the request to receiving the response, releasing resources and basic error handling, this code block has been duplicated elsewhere. Generalize and extract into a separate class